### ISSUE TITLE
DatePicker (Calendar): hidden date parts should always be set to 0

### ISSUE THREAD

**Issue by Peanut4287** [primefaces/primevue#6080]
#### First scenario (DatePicker without `showTime`)

When selecting a day, the component properly sets the values for hours, minutes, seconds and milliseconds to `0` (in local time) on the bound `v-model` `Date` object. See the `date only` log in video.

#### Second scenario (DatePicker with `showTime` but without `showSeconds`)

When changing hours or minutes, the component does not set the values for seconds and milliseconds to `0` on the `Date` object. See 1st `with time` log in video.

If a day is selected in this scenario, the component sets the value for milliseconds to `0` on the `Date` object, but does not set the value for seconds to `0`. See 2nd `with time` log in video.

#### Third scenario (DatePicker with `showTime` and `showSeconds`)

When changing hours, minutes or seconds, the component does not set the values for milliseconds to `0` on the `Date` object. See 1st `with seconds` log in video.

If a day is selected in this scenario, the component sets the value for milliseconds to `0` on the `Date` object. See 2nd `with seconds` log in video.

#### TL;DR

The DatePicker component should always set the values for "smaller", non-configurable parts to `0` when `v-model` is updated.
This means, a DatePicker

- without `showTime` should always set all time parts to `0` (already working)
- with `showTime` but without `showSeconds` should always set seconds and milliseconds to `0`
- with `showTime` and `showSeconds` should always set milliseconds to `0`

https://github.com/user-attachments/assets/e595c212-642a-418c-80bc-84e54084ee97

### Reproducer

https://stackblitz.com/~/github.com/Peanut4287/date-picker-v4-styled

### PrimeVue version

3 & 4

### Vue version

3.x

### Language

TypeScript

### Build / Runtime

Vue CLI App

**Comment by DanielStaroscik** [comment]
I have simillar issue.
In my case o have DatePicker:
`<DatePicker v-model="data.exitTime" hourFormat="24" timeOnly @update:modelValue="computeAbsenceTime(data, index)" />`
And when i change selected time, given Date object has set milliseconds:

![image](https://github.com/user-attachments/assets/c8a7914e-ef1e-4934-a184-e1b0678da77f)

![image](https://github.com/user-attachments/assets/3bd77095-2c17-402d-b0fd-023967961836)

I think i that scenario both seconds and milliseconds should be set to 0.

**Comment by DanielStaroscik** [comment]
Has anything moved on this topic?
